### PR TITLE
[Julia] functions-introduction - replace about.md file with concept files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In [this issue](https://github.com/exercism/v3/issues/2293) we're describing an evolution of the specification of this repo.
These changes include, amongst others, replacing the contents of the `after.md` document with individual concept documents. There are two documents per concept:

1. An `about.md` file containing the description of the concept
1. A `links.json` file containing a set of useful links related to the concept. 

This looks as follows in the repo:

```
<track>
├── concepts
│   ├── <concept>
│   │   ├── about.md
│   │   └── links.json
│   └── ...
```

This PR applies this change to this exercise by: 
                    
1. Copying the existing `after.md` file's contents to the `about.md` file of each concept listed in the exercise's `concepts` section in the `config.json` file.
1. Extract the links from the `after.md` file and put them into the `links.json` file

Before merging this PR, please check that the contents of the concept documents makes sense.
This is especially true when the exercise unlocks multiple concepts, as then the concept documents should be updated to only contain the information relevant to that concept.

Note: to make reviewing easier, the PR has been split into three separate commits. Please squash this PR upon merging.

### CODE REVIEW

File: languages/julia/concepts/functions-introduction/links.json [hunk 1]
```diff
@@ -1 +1,10 @@
...
+[
+    {
+        "url": "https://docs.julialang.org/en/v1/manual/functions/",
+        "description": "functions"
+    },
+    {
+        "url": "https://github.com/invenia/BlueStyle",
+        "description": "blue-style"
+    }
+]
```
Comment (SaschaMann) [inline languages/julia/concepts/functions-introduction/links.json:10]: @ErikSchierboom should this file list _all_ links? Those aren't really standalone resources, they're just linked to give context.

Comment (ErikSchierboom) [reply languages/julia/concepts/functions-introduction/links.json:10]: See the discussion on Slack :)

File: languages/julia/concepts/functions-introduction/links.json [hunk 2]
```diff
@@ -1 +1,10 @@
-[]
+[
+    {
+        "url": "https://docs.julialang.org/en/v1/manual/functions/",
+        "description": "functions"
+    },
+    {
+        "url": "https://github.com/invenia/BlueStyle",
+        "description": "blue-style"
```
Comment (SaschaMann) [inline languages/julia/concepts/functions-introduction/links.json:8]:
```suggestion
        "description": "Blue: a Style Guide for Julia, by Invenia."
```

File: languages/julia/concepts/functions-introduction/links.json [hunk 3]
```diff
@@ -1 +1,10 @@
-[]
+[
+    {
+        "url": "https://docs.julialang.org/en/v1/manual/functions/",
+        "description": "functions"
```
Comment (SaschaMann) [inline languages/julia/concepts/functions-introduction/links.json:4]:
```suggestion
        "description": "Functions, Julia Manual."
```

**Review by SaschaMann — APPROVED** [review]

File: languages/julia/concepts/functions-introduction/links.json (hunk 1 above)
Comment (cmcaine) [reply languages/julia/concepts/functions-introduction/links.json]: This wasn't trivial to follow. The answer in #2499 seems to be "No, not all links".

File: languages/julia/concepts/functions-introduction/links.json [hunk 4]
```diff
@@ -1 +1,10 @@
...
+[
+    {
+        "url": "https://docs.julialang.org/en/v1/manual/functions/",
+        "description": "Functions, Julia Manual."
+    },
+    {
+        "url": "https://github.com/invenia/BlueStyle",
+        "description": "Blue: a Style Guide for Julia, by Invenia."
+    }
+]
```
Comment (SaschaMann) [inline languages/julia/concepts/functions-introduction/links.json:10]:
```suggestion
[
    {
        "url": "https://docs.julialang.org/en/v1/manual/functions/",
        "description": "Julia Manual."
    }
]
```

**Review by cmcaine — APPROVED** [review]

**Pull request merged**